pkg/cni/cilium/cilium.go

```diff
@@ -268,7 +280,7 @@ func ValidateValuesUpdate(newValues, oldValues map[string]any, fieldPath *field.
 func validateImmutableValues(newValues, oldValues map[string]any, fieldPath *field.Path, immutableValues []string) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for _, v := range immutableValues {
-		if !reflect.DeepEqual(newValues[v], oldValues[v]) {
```

Comment:
Can we define this as a

```
// compareMaps is needed because DeepEqual doesn't work for ......
func compareMaps(a, b ........) bool {
   return ......
}
```

Reply:
I say this because this is that type of code that I would come across, wonder "why isn't this just using reflect.DeepEqual?", change it and subtly break something. Explaning why a standard function doesn't apply is vital :)

Reply:
Are you fine with writing a test for the `validateImmutableValues` function instead?

Reply:
How would that look like? What are you then testing and how would that test ensure that the actual code is working?